docs/windows-design.md

```diff
@@ -0,0 +1,255 @@
+ rule should change the source address to the Node internal IP.
+* Resubmit the IP packet to the Conntrack table if it is from the uplink interface in Classifier table. This
+ is to ensure the reply packets from external address to the Pod could be forwarded correctly.
+* Ensure the packet could be handled with NAT configurtaion when entering the ct zone in Conntrack table.
+* Rewrite dMAC with the global virtual MAC(aa:bb:cc:dd:ee:ff) in ConntrackState table if it is from the uplink
+ interface and having the SNAT ct_mark.
+* Resubmit the packets to DNAT table in ConntrackState table if they have the SNAT ct_mark but not from the
+ uplink interface. 
+
+Following is an example for SNAT relevant OpenFlow entries.
+```
+Table0:
+table=0, priority=200, in_port=$uplink, ip actions=load:0x3->NXM_NX_REG0[0..15],resubmit(,30)
+
+Table30:
+table=30, priority=200, ip actions=ct(table=31,zone=65520,nat)
+
+Table31:
+table=31, priority=210,ct_state=-new+trk,ct_mark=0x40,ip, reg0=0x3/0xffff actions= load:aabbccddeeff->NXM_OF_ETH_DST[],resubmit(,40)
+table=31, priority=200, ct_state=-new+trk,ct_mark=0x40,ip actions=resubmit(,40)
+table=31, priority=200, in_port=$uplink,ip actions=output:br-int
+
+Table70:
+// Resubmit the packet to L2ForwardingCalculation table if it is for Service traffic. 
+table=70, priority=200, ip,reg0=0x2/0xffff,ct_mark=0x20 actions=resubmit(,80)
+// Resubmit the packet to L2ForwardingCalculation table if it is traffic from Pod to the Node.
+table=70,priority=190,ip,reg0=0x2/0xffff,nw_dst=$local_nodeIP,actions=resubmit(,80)
+// Resubmit the packet to L2ForwardingCalculation table if it is traffic to the local Pods
+table=70,priority=190,ip,nw_dst=10.10.0.0/24,actions=resubmit(,80)
+// Add SNAT mark if the packet is sent from local Pod to external addresses.
+table=70, priority=0,reg0=0x2/0xffff,ip actions=load:0x1->NXM_NX_REG0[17], resubmit(,90)
+
+Table105:
+table=105, priority=200,ip,reg0=0x20000/0x20000, ct_state=+new+trk, actions=ct(commit,table=110,zone=65520,nat(src=${nodeIP}:10000-20000),exec(load:0x40->NXM_NX_CT_MARK[])))
+
+Table110:
+table=110, priority=200,ip,reg0=0x20000/0x20000 actions=output:gw0
+```
+
+### Use Windows named pipe for internal connections
+Named pipe is used for local connections on Windows Node instead of Unix Domain Socket(UDS). It is used in
+these scenarios:
+* OVSDB connection
+* OpenFlow connection
+* The connection between CNI plugin and CNI server
+ 
+## Antrea Management on Windows
+
+### Antrea Agent Management
+Antrea Agent is running as a process on the Windows Node, but it is managed using a DaemonSet. The utility
+[Rancher Wins](https://github.com/rancher/wins) is used to manage the host process inside the container.
+The configurations for Antrea Agent is configured using configmap, and the environment variables are set
+by kubelet on Windows.
+
+### OVS Management
+OVS is working as Windows Service. Two corresponding services are registered: ovsdb-server and ovs-vswitchd.
```

Comment:
OVS is running as 2 Windows Services: one for ovsdb-server and one for ovs-vswitchd.